test/query_test.exs

```diff
@@ -494,4 +494,20 @@ defmodule QueryTest do
       assert_receive [[:void]], 1000
     end)
   end
+
+  test "async query", context do
+    task1 = async_query("SELECT true, false", [])
+    task2 = async_query("SELECT 42", [])
+
+    %{rows: rows} = Task.await(task1)
```

Comment:
The return value should be {:ok, struct} | {:error, struct}, so may need to refactor how replies are sent. The reply/2 needs to send the actual result of query/4